lib/pod/command/trunk/push.rb

```diff
@@ -119,6 +120,10 @@ def validate_podspec
           unless validator.validated?
             raise Informative, "The spec did not pass validation, due to #{validator.failure_reason}."
           end
+
+          # Let the validator's logic for the swift version
+          # set the value for the trunk JSON uploader
+          @swift_version = validator.swift_version
```

Comment:
We may potentially want to check if the pod uses swift at all here? - swift_version will fall back to '3.0' for an ObjC pod here IIRC because the check is handled earlier in validator.